src/org/isf/medicals/service/MedicalsIoOperations.java

```diff
@@ -87,39 +86,34 @@ public Medical getMedical(
 	
 	/**
 	 * Retrieves all stored {@link Medical}s.
-	 * If a description value is provides the medicals are filtered.
+	 * If a description value is provided the medicals are filtered.
 	 * @param type the medical type description.
-	 * @nameSorted if <code>true</code> return the list in alphatecial order, by code otherwise
+	 * @param nameSorted if <code>true</code> return the list in alphabetical order, by code otherwise
 	 * @return the stored medicals.
 	 * @throws OHServiceException if an error occurs retrieving the stored medicals.
 	 */
 	public ArrayList<Medical> getMedicals(String type, boolean nameSorted) throws OHServiceException {
-		ArrayList<Medical> medicals = null;
-
-		List<Object> parameters = new ArrayList<Object>();
-
-		StringBuilder query = new StringBuilder();
-		query.append("SELECT * FROM MEDICALDSR JOIN MEDICALDSRTYPE ON MDSR_MDSRT_ID_A = MDSRT_ID_A ");
+		List<Medical> medicals;
 		
 		if (type != null) {
 			
```

Comment:
It's only suggestion but maybe it's worth to extract 2 private methods?

Reply:
@tomek what do you mean here?

Reply:
Regarding to clean code we should avoid multiple nested ifs so you can extract 2 private methods getWithType and get WithoutType. For instance sth like type != null ? getWithType(type) : getWithoutType()

Reply:
@tomek39856 I get you now and I agree, if it is not too overload for @marcoebbinghaus, generally to change many things in one PR, also with good improvments but out of the issue's scope itself, leads to merge errors. If the change can be done both easily and ensuring tests and gui, it's ok for me.